File: gitdot-web/packages/api/src/endpoint/review/add-reviewer.ts
import { z } from "zod";
import { ReviewerResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const AddReviewerRequest = z.object({
  user_name: z.string(),
});
export type AddReviewerRequest = z.infer<typeof AddReviewerRequest>;

export const AddReviewerResponse = ReviewerResource;
export type AddReviewerResponse = z.infer<typeof AddReviewerResponse>;

export const AddReviewer = {
  path: "/repository/{owner}/{repo}/review/{number}/reviewer",
  method: "POST",
  request: AddReviewerRequest,
  response: AddReviewerResponse,
} as const satisfies Endpoint;
export type AddReviewer = typeof AddReviewer;

File: gitdot-api/src/endpoint/review/add_reviewer.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::review::ReviewerResource};

pub struct AddReviewer;

impl Endpoint for AddReviewer {
    const PATH: &'static str = "/repository/{owner}/{repo}/review/{number}/reviewer";
    const METHOD: http::Method = http::Method::POST;

    type Request = AddReviewerRequest;
    type Response = AddReviewerResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct AddReviewerRequest {
    pub user_name: String,
}

pub type AddReviewerResponse = ReviewerResource;
